rl4co/utils/ops.py

```diff
@@ -93,7 +93,31 @@ def get_tour_length(ordered_locs):
     ordered_locs_next = torch.roll(ordered_locs, 1, dims=-2)
     return get_distance(ordered_locs_next, ordered_locs).sum(-1)
 
+@torch.jit.script
+def get_open_tour_length(ordered_locs):
+    """Compute the total tour distance for a batch of ordered tours.
+    Computes the L2 norm between each pair of consecutive nodes in the tour and sums them up.
+
+    Args:
+        ordered_locs: Tensor of shape [batch_size, num_nodes, 2] containing the ordered locations of the tour
+    """
+    ordered_locs_next = torch.roll(ordered_locs, 1, dims=-2)
+    
+    segment_lengths = ((ordered_locs_next-ordered_locs)**2).sum(-1).sqrt()
+
+    # Get the first value of ordered_locs
+    first_loc = ordered_locs[:, 0, :][:,None,:].expand(ordered_locs_next.shape)
+
+    # Check the ids where the location is the same as the first value
+    same_loc_ids = torch.all(ordered_locs_next == first_loc, dim=-1)
+
+    # for open VRP, the distance between last customer and the depot is not counted
+    segment_lengths[same_loc_ids] = 0
```

Comment:
Is it just the last customer for which we don't need to return to the depot? Why don't you do something like the following, then:
```
def get_open_tour_length(ordered_locs):
    ordered_locs_next = torch.roll(ordered_locs, -1, dims=-2)
    return get_distance(ordered_locs[:, :-1, :], ordered_locs_next[:, :-1, :]).sum(-1)
```
Notice how (1) in torch.roll `shifts=-1` instead of 1 (i.e. we shift the sequence to the left), and (2) in `get_distance()` the order of the two parameters is also exchanged. But just omitting the last entry in both tensors you're not travelling back to the depot.

Reply:
I have not actually run the code, but I did something similar (without the indexing) at a different point, so in principle it shoudl work :)

Reply:
I think if it's the last customer for each route, we may still need to mask out the middle lengths.
I think that in your code snippet, an action sequence like `[0, 3, 5, 2, 0, 1, 4, 0]` where 0 is the depot will consider only the last route (0, 1, 4)  as open but not (0, 3, 5, 2)

Reply:
Aah, ok, so all individual routes are open. Then my solution doesn't make sense :D

Reply:
Yes, according the the definition, it seems to be the case (e.g. [here](https://www.sciencedirect.com/science/article/abs/pii/S0305054805003515#:~:text=to%20the%20depot.-,In%20the%20open%20vehicle%20routing%20problem%20(OVRP)%2C%20a%20vehicle%20does%20not%20return%20to%20the%20depot%20after%20servicing%20the%20last%20customer%20on%20a%20route.,-Each%20route%20in))